api/core/v1beta2/machineset_types.go

```diff
@@ -33,6 +33,27 @@ const (
 	// MachineSetFinalizer is the finalizer used by the MachineSet controller to
 	// ensure ordered cleanup of corresponding Machines when a Machineset is being deleted.
 	MachineSetFinalizer = "cluster.x-k8s.io/machineset"
+
+	// MachineSetMoveMachinesToAnnotationName is an internal annotation added by the MD controller to the oldMS
+	// when it should scale down by moving machines that can be updated in-place to the newMS instead of deleting them.
+	// Note: This annotation is used to perform a two-ways check before moving a machine from oldMS to newMS:
+	//
+	//	"oldMS must have: move to newMS" and "newMS must have: receive replicas from oldMS"
+	MachineSetMoveMachinesToAnnotationName = "in-place-updates.internal.cluster.x-k8s.io/move-machines-to-machineset"
+
+	// MachineSetReceiveMachinesFromAnnotationName is an internal annotation added by the MD controller to the newMS
+	// when it should receive replicas from an oldMS as a first step of an in-place upgrade operation.
+	// Note: This annotation is used to perform a two-ways check before moving a machine from oldMS to newMS:
+	//
+	//	"oldMS must have: move to newMS" and "newMS must have: receive replicas from oldMS"
+	MachineSetReceiveMachinesFromAnnotationName = "in-place-updates.internal.cluster.x-k8s.io/receive-machines-from-machinesets"
+
+	// MachineSetAcknowledgeMoveAnnotationName is an internal annotation with a list of machines added by the MD controller
+	// to a MachineSet when it acknowledges a machine pending acknowled after being moved from an oldMS.
+	// A machine is dropped from this annotation as soon as pending-acknowledge-move is removed from the machine;
+	// the annotation is dropped when empty.
+	// Note: this annotation is used in pair with replicasMachineDeploymentAcknowledgeAnnotation and replicaPendingMachineDeploymentAcknowledgeAnnotation.
+	MachineSetAcknowledgeMoveAnnotationName = "in-place-updates.internal.cluster.x-k8s.io/acknowledge-move"
```

Comment:
```suggestion
	MachineSetAcknowledgeMoveAnnotationName = "in-place-updates.internal.cluster.x-k8s.io/acknowledged-move"
```
Should this be acknowledged move? I think this is the MD controller telling the MS controller that this Machine was acknowledged by the MD controller.

(not the MD controller telling the MS controller to acknowledge the Machine)